ocaml/typing/ctype.ml

```diff
@@ -2221,14 +2221,12 @@ let () =
   Env.constrain_type_jkind := constrain_type_jkind
 
 let check_type_externality env ty ext =
-  let check_sub ty_jkind =
-    Jkind.(Externality.le (get_externality_upper_bound ty_jkind) ext)
+  let upper_bound =
```

Comment:
Probably worth memoizing this. This function appears to be called a lot in transl. (This was true of my old implementation, too.)

Reply:
The lifetime of this thing is just the next function call, so I strongly expect this to always be a minor allocation and therefore ~free.  Memoizing things like the core kinds in jkind.ml makes more sense, because their lifetime might be all of typechecking.  But maybe my intuition is wrong here?

Reply:
(The above remark is wrong, we might end up storing the thing in a tyvar as a result of this check.  I will change this,)

Reply:
Actually I believe the first remark (this uses `check_type_jkind`, not `constrain_type_jkind`, so it won't end up in the tyvar) - sorry for the noise, a lot going on today.

Reply:
Yes, OK.